### PR TITLE
flang support for pointers to scalar variables during debug info generation

### PR DESCRIPTION
**Pull request by bhuvanendrakumarn**
This fix handles pointer to scalar variable (of symbol type:ST_VAR) during debug info generation. It address 2 issues.

1. flang is not emitting DW_AT_name attribute for pointer variable DIE, hence inside GDB, "print cqe_int_allocatable_scalar" command fails as shown below
2. flang is emitting DW_TAG_array_type for a pointer variable instead of DW_TAG_pointer_type

Test case: 

program test_cqe
integer, allocatable :: cqe_int_allocatable_scalar
cqe_int_allocatable_scalar=5
print *,cqe_int_allocatable_scalar
print *,"End of printing"
end program test_cqe

sample GDB session:

- without fix :
(gdb) p cqe_int_allocatable_scalar
No symbol "cqe_int_allocatable_scalar" in current context.

- with fix :
(gdb) p cqe_int_allocatable_scalar
$1 = (PTR TO -> ( integer )) 0x21be90
(gdb) p *cqe_int_allocatable_scalar
$2 = 5

### CODE REVIEW

**Comment by sscalpone** [comment]
@schweitzpgi FYI

**Comment by bhuvanendrakumarn** [comment]
gentle remainder for upstream review request

**Comment by bhuvanendrakumarn** [comment]
@jiel-nv, Thanks a lot for the review.
@schweitzpgi, Gentle reminder for review request.

File: tools/flang2/flang2exe/cgmain.cpp [hunk 1]
```diff
@@ -10493,6 +10493,21 @@ create_global_initializer(GBL_LIST *gitem, const char *flag_str,
   gitem->global_def = gname;
 }
 
+/**
+   \brief Check if sptr is the midnum of a scalar and scalar has POINTER/ALLOCATABLE attribute
+   \param sptr  A symbol
+ */
```
Comment (schweitzpgi) [inline tools/flang2/flang2exe/cgmain.cpp:10499]: You probably don't need the doxygen comment on both the decl and the defn.

**Comment by bhuvanendrakumarn** [comment]
@jiel-nv, @schweitzpgi, Thanks a lot for the review.

**Comment by bhuvanendrakumarn** [comment]
Hi all,

>Sent: Wednesday, June 10, 2020 11:26 AM
>Subject: Re: [flang-compiler/flang] flang support for pointers to scalar variables during debug info generation (#887)
>@bhuvanendrakumarn pushed 1 commit.
•	38f0962 flang support for Assumed length string type during debug info generation

i have accidentally pushed some other changes to feature-pointertoscalar branch which was created for approved PR (887). is there anyway i can revert it back under feature-pointertoscalar branch?

any help/suggestion in this regard will be helpful

**Comment by bryanpkc** [comment]
@bhuvanendrakumarn Is this PR still relevant? Would you like to update it, or close it?

**Comment by bryanpkc** [comment]
@bhuvanendrakumarn Should we close this PR?

**Comment by bhuvanendrakumarn** [comment]
@bryanpkc Hi, the fix for this issue(#887) is already merged. hence closing this ticket

**Comment by bhuvanendrakumarn** [comment]
Fix for this issue already merged

**Pull request closed without merging**